sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala

```diff
@@ -154,15 +154,17 @@ private class ChangeFeedBatch
         if (!metadataLog.add(0, latestOffsetJson)) {
           val existingLatestOffset = metadataLog.get(0).get
 
-          val msg = s"Cannot update latest offset at location '$latestOffsetLocation' for batchId: $batchId " +
-            s"-> existing latestOffset: '$existingLatestOffset' failed to persist " +
-            s"new latestOffset: '$latestOffsetJson'."
-
           if (existingLatestOffset != latestOffsetJson) {
-            log.logError(msg)
+            val msg = s"Cannot update latest offset at location '$latestOffsetLocation' for batchId: $batchId " +
+              s"-> existing latestOffset: '$existingLatestOffset' failed to persist " +
+              s"new latestOffset: '$latestOffsetJson'- will continue with existing latest offset."
+            log.logWarning(msg)
 
-            throw new IllegalStateException(msg)
+            latestOffset = ChangeFeedOffset.fromJson(existingLatestOffset)
```

Comment:
should we also validate the collectionRid is same?

Reply:
Not sure it is really needed - but won't hurt. Added it.